File: codes/0686-Repeated-String-Match/Java/main3.java
class Solution {
    public int repeatedStringMatch(String a, String b) {
        int n1 = a.length(), n2 = b.length(), cnt = 1;
        String t = a;
        while (t.length() < n2) {
            t += a;
            ++cnt;
        }
        if (t.contains(b)) return cnt;
        t += a;
        return (t.contains(b)) ? cnt + 1 : -1;
    }
}

File: codes/0565-Array-Nesting/cpp/main3.cpp
class Solution
{
public:
    int arrayNesting(vector<int> &nums)
    {
        int n = nums.size(), res = 0;
        for (int i = 0; i < n; ++i)
        {
            int cnt = 1;
            while (nums[i] != i)
            {
                swap(nums[i], nums[nums[i]]);
                ++cnt;
            }
            res = max(res, cnt);
        }
        return res;
    }
};